pkg/acquisition/modules/syslog/internal/parser/rfc3164/parse.go

```diff
@@ -0,0 +1,256 @@
+package rfc3164
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/crowdsecurity/crowdsec/pkg/acquisition/modules/syslog/internal/parser/utils"
+)
+
+type RFC3164Option func(*RFC3164)
+
+type RFC3164 struct {
+	PRI       int
+	Timestamp time.Time
+	Hostname  string
+	Tag       string
+	Message   string
+	PID       string
+	//
+	len            int
+	position       int
+	buf            []byte
+	useCurrentYear bool //If no year is specified in the timestamp, use the current year
+	strictHostname bool //If the hostname contains invalid characters or is not an IP, return an error
+}
+
+const PRI_MAX_LEN = 3
+
+//Order is important: format with the most information must be first because we will stop on the first match
+var VALID_TIMESTAMPS = []string{
+	time.RFC3339,
+	"Jan 02 15:04:05 2006",
+	"Jan 2 15:04:05 2006",
+	"Jan 02 15:04:05",
+	"Jan 2 15:04:05",
+}
+
+func WithCurrentYear() RFC3164Option {
+	return func(r *RFC3164) {
+		r.useCurrentYear = true
+	}
+}
+
+func WithStrictHostname() RFC3164Option {
+	return func(r *RFC3164) {
+		r.strictHostname = true
+	}
+}
+
+func (r *RFC3164) parsePRI() error {
+
+	pri := 0
+
+	if r.buf[r.position] != '<' {
+		return fmt.Errorf("PRI must start with '<'")
+	}
+
+	r.position++
+
+	for r.position < r.len {
+		c := r.buf[r.position]
+		if c == '>' {
+			r.position++
+			break
+		}
+		if c < '0' || c > '9' {
+			return fmt.Errorf("PRI must be a number")
+		}
+		pri = pri*10 + int(c-'0')
+		r.position++
+	}
+
+	if pri > 999 {
+		return fmt.Errorf("PRI must be up to 3 characters long")
+	}
+
+	if r.position == r.len && r.buf[r.position-1] != '>' {
+		return fmt.Errorf("PRI must end with '>'")
+	}
+
+	r.PRI = pri
+	return nil
+}
+
+func (r *RFC3164) parseTimestamp() error {
+	validTs := false
+	for _, layout := range VALID_TIMESTAMPS {
+		tsLen := len(layout)
+		if r.position+tsLen > r.len {
```

Comment:
not sure it's  always right. Can it happen that an actual timestamp is shorter than the layout string ? what if hours are < 10 ? (not sure tho)